pkg/kubelet/volumes.go

```diff
@@ -134,6 +142,25 @@ func (kl *Kubelet) mountExternalVolumes(pod *api.Pod) (kubecontainer.VolumeMap,
 		if err != nil {
 			return nil, err
 		}
+		if hasFSGroup && builder.SupportsOwnershipManagement() {
```

Comment:
> So the question is whether SupportsFooBar() should internally check read-only or whether the caller should check it. Given that IsReadOnly() is an existing public interface, I think we should make callers check it. Some volumes might internally check it or some other condition, but it makes sense to me to put it here for all volumes. What think?

I'll make this change to #15323. If we come up with a use-case where a volume wants to be read only yet still be relabled and owned by FSGroup we can revisit it then